File: snapo-app-mac/Snap-O/NetworkInspector/NetworkInspectorStatusPresentation.swift
import Foundation
import SwiftUI

enum NetworkInspectorStatusPresentation {
  static func color(for code: Int) -> Color {
    switch code {
    case 200 ..< 300:
      .green
    case 400 ..< 600:
      .red
    case 300 ..< 400:
      .orange
    case 100 ..< 200:
      .secondary
    default:
      .secondary
    }
  }

  static func displayName(for code: Int) -> String {
    if let override = overrides[code] {
      return "\(code) \(override)"
    }

    let raw = HTTPURLResponse.localizedString(forStatusCode: code)
    let trimmed = raw.trimmingCharacters(in: .whitespacesAndNewlines)

    let descriptor: String = if trimmed.isEmpty || trimmed == "unknown" {
      "Done"
    } else if trimmed.lowercased() == "ok" {
      "OK"
    } else if trimmed.lowercased() == "no error" {
      "OK"
    } else {
      trimmed.capitalized
    }

    return "\(code) \(descriptor)"
  }

  private static let overrides: [Int: String] = [
    200: "OK",
    201: "Created",
    202: "Accepted",
    204: "No Content",
    301: "Moved Permanently",
    302: "Found",
    304: "Not Modified",
    307: "Temporary Redirect",
    308: "Permanent Redirect",
    400: "Bad Request",
    401: "Unauthorized",
    403: "Forbidden",
    404: "Not Found",
    405: "Method Not Allowed",
    409: "Conflict",
    410: "Gone",
    422: "Unprocessable Entity",
    429: "Too Many Requests",
    500: "Internal Server Error",
    501: "Not Implemented",
    502: "Bad Gateway",
    503: "Service Unavailable",
    504: "Gateway Timeout"
  ]
}

File: snapo-desktop-compose/src/main/kotlin/com/openai/snapo/desktop/inspector/NetworkInspectorStatusPresentation.kt
package com.openai.snapo.desktop.inspector

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import com.openai.snapo.desktop.ui.theme.SnapOAccents

object NetworkInspectorStatusPresentation {
    @Composable
    fun color(code: Int): Color {
        val accents = SnapOAccents.current()
        return when (code) {
            in 200..299 -> accents.success
            in 400..599 -> accents.error
            in 300..399 -> accents.warning
            in 100..199 -> accents.info
            else -> accents.info
        }
    }

    fun displayName(code: Int): String {
        val override = overrides[code]
        if (override != null) return "$code $override"
        return "$code Done"
    }

    private val overrides: Map<Int, String> = mapOf(
        200 to "OK",
        201 to "Created",
        202 to "Accepted",
        204 to "No Content",
        301 to "Moved Permanently",
        302 to "Found",
        304 to "Not Modified",
        307 to "Temporary Redirect",
        308 to "Permanent Redirect",
        400 to "Bad Request",
        401 to "Unauthorized",
        403 to "Forbidden",
        404 to "Not Found",
        405 to "Method Not Allowed",
        409 to "Conflict",
        410 to "Gone",
        422 to "Unprocessable Entity",
        429 to "Too Many Requests",
        500 to "Internal Server Error",
        501 to "Not Implemented",
        502 to "Bad Gateway",
        503 to "Service Unavailable",
        504 to "Gateway Timeout",
    )
}
